app/src/types/user.js

```diff
@@ -2,22 +2,80 @@
 
 "use strict";
 
-export type User = {
-  +created_at: number,
-  +token: string,
-  +spid_idp: string,
-  +fiscal_code: string,
-  +name: string,
-  +familyname: string,
-  +spidcode?: string,
-  +gender?: string,
-  +mobilephone?: string,
-  +email?: string,
-  +address?: string,
-  +expirationdate?: string,
-  +digitaladdress?: string,
-  +countyofbirth?: string,
-  +dateofbirth?: string,
-  +idcard?: string,
-  +placeofbirth?: string
-};
+import * as t from "io-ts";
+
+const UserModel = t.intersection([
+  t.type({
+    created_at: t.number,
+    token: t.string,
+    spid_idp: t.string,
+    fiscal_code: t.string,
+    name: t.string,
+    familyname: t.string
+  }),
+  t.partial({
+    created_at: t.number,
+    token: t.string,
+    spid_idp: t.string,
+    fiscal_code: t.string,
+    name: t.string,
+    familyname: t.string,
+    spidcode: t.string,
+    gender: t.string,
+    mobilephone: t.string,
+    email: t.string,
+    address: t.string,
+    expirationdate: t.string,
+    digitaladdress: t.string,
+    countyofbirth: t.string,
+    dateofbirth: t.string,
+    idcard: t.string,
+    placeofbirth: t.string
+  })
+]);
+
+export type User = t.TypeOf<typeof UserModel>;
+
+/**
+ * Converts the user added to the request by Passport to an User object.
+ *
+ * @param from
+ * @returns {User}
+ */
+export function extractUserFromRequest(from: express$Request): User {
+  const reqWithUser = ((from: Object): { user: User });
+
+  // eslint-disable-next-line no-unused-vars
+  return t.validate(reqWithUser.user, UserModel).fold(_ => {
+    return createEmptyUser();
+  }, t.identity);
+}
+
+/**
+ * Converts a json string to an User object.
+ *
+ * @param from
+ * @returns {User}
+ */
+export function extractUserFromJson(from: string): User {
+  // eslint-disable-next-line no-unused-vars
+  return t.validate(JSON.parse(from), UserModel).fold(_ => {
+    return createEmptyUser();
```

Comment:
not sure about this logic, if the JSON doesn't represent a valid `UserModel` why we create an empty user? shouldn't it just throw an error? or better `extractUserFromJson` should return and `Option` or an `Either`